src/DaisyFx.NCrontab/NCrontabSource.cs

```diff
@@ -23,9 +23,11 @@ public override async Task ExecuteAsync(SourceNextDelegate<Signal> next, Cancell
             {
                 var nextOccurrence = _cron.GetNextOccurrence(DateTime.UtcNow);
                 var timeUntilNextOccurrence = nextOccurrence - DateTime.UtcNow;
-                var waitUntilNextOccurrenceTask = Task.Delay(timeUntilNextOccurrence, cancellationToken);
+                // Round up to nearest whole second as waiting with a sub-second resolution is unreliable
+                var delay = TimeSpan.FromSeconds(Math.Ceiling(timeUntilNextOccurrence.TotalSeconds));
+                var waitUntilNextOccurrenceTask = Task.Delay(delay, cancellationToken);
 
-                _logger.Waiting(timeUntilNextOccurrence);
+                _logger.Waiting(delay);
                 await waitUntilNextOccurrenceTask;
```

Comment:
Was/is there a reason for us not directly awaiting the `Task.Delay` like this:
```
var delay = TimeSpan.FromSeconds(Math.Ceiling(timeUntilNextOccurrence.TotalSeconds));
_logger.Waiting(delay);
await Task.Delay(delay, cancellationToken);
```

Reply:
Fixed